File: data/compilers_comparation/src/examples/first_example/src/func42.c
#include "first_example.h" 
array_t* func42(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array18 = func43(&params0, loopsFactor);
      DEBUG_RETURN(array18->id);
      free(params0.data);
      array18->refC--;
      if(array18->refC == 0) {
         free(array18->data);
         free(array18);
         DEBUG_FREE(array18->id);
      }
   }
   else {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array20 = func44(&params0, loopsFactor);
      DEBUG_RETURN(array20->id);
      free(params0.data);
      array20->refC--;
      if(array20->refC == 0) {
         free(array20->data);
         free(array20);
         DEBUG_FREE(array20->id);
      }
   }
   array_t_param params0;
   params0.size = 0;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   array_t* array22 = func45(&params0, loopsFactor);
   DEBUG_RETURN(array22->id);
   free(params0.data);
   return array22;
}

File: data/multilang_comparison/program_nim/src/func42.nim
proc func42*(vars: ptr ArrayTParam, PATH0: uint64, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   if (PATH0 and 1'u64) != 0:
      discard
      var params0: ArrayTParam
      params0.size = 0
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      var array71: ptr ArrayT = func43(addr params0, getPath(), loopsFactor)
      debugReturn(array71.id)
      dealloc(params0.data)
      dec array71.refC
      if array71.refC == 0:
         dealloc(array71.data)
         dealloc(array71)
         debugFree(array71.id)
   else:
      discard
      var params0: ArrayTParam
      params0.size = 0
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      var array80: ptr ArrayT = func44(addr params0, getPath(), loopsFactor)
      debugReturn(array80.id)
      dealloc(params0.data)
      dec array80.refC
      if array80.refC == 0:
         dealloc(array80.data)
         dealloc(array80)
         debugFree(array80.id)
   var params0: ArrayTParam
   params0.size = 0
   params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
   var array85: ptr ArrayT = func45(addr params0, getPath(), loopsFactor)
   debugReturn(array85.id)
   dealloc(params0.data)
   return array85

